src/Core/Module/ModuleRepository.php

```diff
@@ -0,0 +1,256 @@
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Module;
+
+use Doctrine\Common\Cache\CacheProvider;
+use Module as ModuleLegacy;
+use PrestaShop\PrestaShop\Adapter\Entity\Shop;
+use PrestaShop\PrestaShop\Adapter\HookManager;
+use PrestaShop\PrestaShop\Adapter\Module\AdminModuleDataProvider;
+use PrestaShop\PrestaShop\Adapter\Module\Module;
+use PrestaShop\PrestaShop\Adapter\Module\ModuleDataProvider;
+use Symfony\Component\Finder\Finder;
+
+class ModuleRepository implements ModuleRepositoryInterface
+{
+    private const MODULE_ATTRIBUTES = [
+        'warning',
+        'name',
+        'tab',
+        'displayName',
+        'description',
+        'author',
+        'author_address',
+        'limited_countries',
+        'need_instance',
+        'confirmUninstall',
+    ];
+
+    /** @var Finder */
+    private $finder;
+
+    /** @var ModuleDataProvider */
+    private $moduleDataProvider;
+
+    /** @var AdminModuleDataProvider */
+    private $adminModuleDataProvider;
+
+    /** @var HookManager */
+    private $hookManager;
+
+    /** @var CacheProvider */
+    private $cacheProvider;
+
+    /** @var string */
+    private $modulePath;
+
+    /** @var array|null */
+    private $installedModules;
+
+    public function __construct(
+        ModuleDataProvider $moduleDataProvider,
+        AdminModuleDataProvider $adminModuleDataProvider,
+        CacheProvider $cacheProvider,
+        HookManager $hookManager,
+        string $modulePath
+    ) {
+        $this->finder = new Finder();
+        $this->moduleDataProvider = $moduleDataProvider;
+        $this->adminModuleDataProvider = $adminModuleDataProvider;
+        $this->cacheProvider = $cacheProvider;
+        $this->hookManager = $hookManager;
+        $this->modulePath = $modulePath;
+    }
+
+    public function getList(): array
+    {
+        $modules = [];
+        $modulesDirsList = $this->finder->directories()
+            ->in($this->modulePath)
+            ->depth('== 0')
+            ->exclude(['__MACOSX'])
+            ->ignoreVCS(true);
+
+        foreach ($modulesDirsList as $moduleDir) {
+            $moduleName = $moduleDir->getFilename();
+            if (null === $this->getModulePath($moduleName)) {
+                continue;
+            }
+
+            $modules[] = $this->getModule($moduleName);
+        }
+
+        return $this->mergeWithModulesFromHook($modules);
+    }
+
+    public function getInstalledModules(): array
+    {
+        return array_filter($this->getList(), function (Module $module) {
+            return $module->database->get('installed') === true;
+        });
+    }
+
+    public function getConfigurableModules(): array
+    {
+        return array_filter($this->getList(), function (Module $module) {
+            return $module->hasValidInstance() && !empty($module->getInstance()->warning);
+        });
+    }
```

Comment:
Only a suggestion to keep the code easier to read, and to exploit `ModuleCollection` to its full potential:
- All methods here should return `ModuleCollection` instead of `ModuleInterface[]`
- ModuleCollection should implement a filter method: `public function filter(callable $filterCallback): self`

```suggestion
    public function getConfigurableModules(): ModuleCollection
    {
        return $this->getList()->filter(static function (Module $module): bool {
            return $module->hasValidInstance() && !empty($module->getInstance()->warning);
        });
    }
```